src/vm/values/custom/vversion.nim

```diff
@@ -0,0 +1,57 @@
+#=======================================================
+# Arturo
+# Programming Language + Bytecode VM compiler
+# (c) 2019-2022 Yanis Zafirópulos
+#
+# @file: vm/values/custom/vversion.nim
+#=======================================================
+
+## The internal `:version` type
+
+#=======================================
+# Libraries
+#=======================================
+
+#=======================================
+# Types
+#=======================================
+
+type
+    VVersion* = ref object
+        major*   : int
+        minor*   : int
+        patch*   : int
+        extra*   : string
+
+#=======================================
+# Overloads
+#=======================================
+
+func `==`*(a, b: VVersion): bool {.inline,enforceNoRaises.} =
+    return a.major == b.major and 
+           a.minor == b.minor and 
+           a.patch == b.patch and 
+           a.extra == b.extra
+
+func `<`*(a, b: VVersion): bool {.inline,enforceNoRaises.} =
```

Comment:
```nim
func `<`*(a, b: VVersion): bool {.inline,enforceNoRaises.} =
    let
        majorEqual = a.major == b.major
        majorMinorEqual = a.minor == b.minor and majorEqual
    a.major < b.major or (majorEqual and a.minor < b.minor) or (majorMinorEqual and a.patch < b.patch)
  ```

Reply:
I re-wrote this (and `>`) a bit differently, inspired by your version, but also trying to avoid forward tests that may be unnecessary (again, not that it matter so much, since they are Version values; but, still, every bit might matter at some point...) 😉